### PR TITLE
[PIN-1291] migration of PUT /eservices/{eServiceId} from catalog-process

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -3606,6 +3689,26 @@ components:
...
+        - attributes
+      properties:
+        name:
+          type: string
+          minLength: 5
+          maxLength: 60
+        description:
+          type: string
+          minLength: 10
+          maxLength: 250
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: fields constraints should be only on the process

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 2]
```diff
@@ -365,4 +365,19 @@ final case class EServicesApiServiceImpl(
...
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCreatedResource: ToEntityMarshaller[CreatedResource]
+  ): Route = {
+    val result: Future[CreatedResource] =
+      catalogProcessService.updateEServiceById(eServiceId, updateEServiceSeed.toProcess)(contexts).map(_.toApi)
+
+    onComplete(result) {
+      handleError(s"Error updating eservice with Id: $eServiceId") orElse { case Success(eservice) =>
+        createEService200(eservice)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]:
```suggestion
        updateEService200(eservice)
```

**Review by galales — CHANGES_REQUESTED** [review]

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala [hunk 3]
```diff
@@ -64,4 +70,16 @@ class CatalogProcessServiceImpl(catalogProcessUrl: String, blockingEc: Execution
...
+    contexts: Seq[(String, String)]
+  ): Future[EService] = withHeaders { (bearerToken, correlationId, ip) =>
+    val request: ApiRequest[EService] =
+      api.updateEServiceById(
+        xCorrelationId = correlationId,
+        eServiceId = eServiceId.toString,
+        updateEServiceSeed = updateEServiceSeed,
+        xForwardedFor = ip
+      )(BearerToken(bearerToken))
+    invoker.invoke(request, s"Updating EService with $eServiceId")
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala:218]: @galales won't this require a bit of error handling?

**Review by TonioGela — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala (hunk 3 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala:218]: the BFF is a bit different about the error handling, it forwards everything it receives, http status included.
In this service it is safe to not handle each error response 👍

**Review by TonioGela — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 4]
```diff
@@ -2111,6 +2111,89 @@ paths:
...
+        - eservices
+      summary: Updates EService general information
+      operationId: updateEServiceById
+      parameters:
+        - name: eServiceId
+          in: path
+          description: The E-Service id to update
+          required: true
+          schema:
+            type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml:2624]:
```suggestion
            type: string
            format: uuid
```

**Review by galales — CHANGES_REQUESTED** [review]

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**